eclair-core/src/main/scala/fr/acinq/eclair/message/OnionMessages.scala

```diff
@@ -44,10 +44,11 @@ object OnionMessages {
 
   def buildRoute(blindingSecret: PrivateKey,
                  intermediateNodes: Seq[IntermediateNode],
-                 destination: Destination): Sphinx.RouteBlinding.BlindedRoute = {
+                 destination: Destination,
+                 originKey: Option[PrivateKey] = None): Try[Sphinx.RouteBlinding.BlindedRoute] = Try{
```

Comment:
The reason it is optional is that we don't always know who is the origin node. It could be us if we're building a route to use it to send a message but it could be anyone if we're building a reply path.

Regarding exceptions, your alternative is just more verbose. I think `Try { myOption.get }` is much cleaner than `myOption match { case Some(x) => Success(x) case None => Failure(new IllegalArgumentException("empty option")) }`.

Reply:
> The reason it is optional is that we don't always know who is the origin node.

But we don't care about the origin node, we only care about the case where _we_ are the origin node, right? The only thing we want to check here is that if we (ie `nodeParams.nodeId`) are the introduction node of the destination's blinded route, we should decrypt one hop.

> I think Try { myOption.get } is much cleaner

`myOption.get` throws an exception, which captures a stack trace, which is expensive and should simply always be avoided, this is one of the most repeated coding best practice: never throw exceptions when you can easily avoid it.